### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by miigotu**
Fixes #

Proposed changes in this pull request:
-
-
-

- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/SickChill/SickChill/blob/master/.github/CONTRIBUTING.md)

## Summary by CodeRabbit

- **Refactor**
  - Updated code formatting rules for consistency across various file types.
  - Modified pre-commit rules to relax end-of-file checks.

- **Documentation**
  - Updated the Changelog with recent changes and fixes.
  - Revised URLs for Changelog and Network Timezones documentation.

- **New Features**
  - Added new web interface components and routes for SickChill.

- **Dependencies**
  - Updated numerous package versions to enhance functionality and security.
  - Removed support for Python 3.7 and added support for Python 3.10.

- **Configuration**
  - Adjusted GitHub Actions workflows for improved automation.
  - Updated Dependabot and funding configuration for better maintenance and support.

- **Tooling**
  - Introduced Qodana analysis configuration for code quality checks.

### CODE REVIEW

File: .github/workflows/check_release.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          virtualenvs-path: .venv
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: false
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: "poetry"
+          cache-dependency-path: pyproject.toml
```
[the remark this answers is not in the document]
Comment (miigotu) [reply .github/workflows/check_release.yml]: cache and cache-dependency-path are inputs since the v3 release

**Comment by miigotu** [comment]
!LGTM

**Comment by miigotu** [comment]
LGTM

**Comment by miigotu** [comment]
```yaml annotate
# The name of the workflow as it will appear in the "Actions" tab of the GitHub repository.
name: Post welcome comment
# The `on` keyword lets you define the events that trigger when the workflow is run.
on:
  # Add the `pull_request` event, so that the workflow runs automatically
  # every time a pull request is created.
  pull_request:
    types: [opened]
# Modifies the default permissions granted to `GITHUB_TOKEN`.
permissions:
  pull-requests: write
# Defines a job with the ID `build` that is stored within the `jobs` key.
jobs:
  build:
    name: Post welcome comment
    # Configures the operating system the job runs on.
    runs-on: ubuntu-latest
    # The `run` keyword tells the job to execute a command on the runner.
    steps:
      - run: gh pr comment $PR_URL --body "Welcome to the repository!"
        env:
          GITHUB_TOKEN: $
          PR_URL: $
```

File: sickchill/gui/slick/views/history.mako [hunk 2]
```diff
@@ -15,30 +15,29 @@
...
-                        <option value="25" ${('', 'selected="selected"')[limit == 25]}>25</option>
-                        <option value="50" ${('', 'selected="selected"')[limit == 50]}>50</option>
-                        <option value="100" ${('', 'selected="selected"')[limit == 100]}>100</option>
-                        <option value="250" ${('', 'selected="selected"')[limit == 250]}>250</option>
-                        <option value="500" ${('', 'selected="selected"')[limit == 500]}>500</option>
-                        <option value="750" ${('', 'selected="selected"')[limit == 750]}>750</option>
-                        <option value="1000" ${('', 'selected="selected"')[limit == 1000]}>1000</option>
-                        <option value="0" ${('', 'selected="selected"')[limit == 0  ]}>All</option>
+                        % for val in [10, 25, 50, 100, 250, 500, 750, 1000, 0]:
+                            <option value="${val}}" ${selected(limit == val)}>${(0, _(All))[val == 0]}</option>
```
Comment (BKSteve) [inline sickchill/gui/slick/views/history.mako:26]: creates `All` not defined mako error when displaying History page.

File: sickchill/gui/slick/views/inc_qualityChooser.mako [hunk 3]
```diff
@@ -19,7 +19,7 @@ selected = None
         <select id="qualityPreset" name="quality_preset" class="form-control input-sm input100" title="qualityPreset">
             <option value="0">Custom</option>
             % for curPreset in qualityPresets:
-                <option value="${curPreset}" ${('', 'selected="selected"')[curPreset == overall_quality]} ${('', 'style="padding-left: 15px;"')[qualityPresetStrings[curPreset].endswith("0p")]}>${qualityPresetStrings[curPreset]}</option>
+                <option value="${curPreset}" ${selected(curPreset == overall_quality)} ${('', 'style="padding-left: 15px;"')[qualityPresetStrings[curPreset].endswith("0p")]}>${qualityPresetStrings[curPreset]}</option>
```
Comment (BKSteve) [inline sickchill/gui/slick/views/inc_qualityChooser.mako:21]: TypeError: 'NoneType' object is not subscriptable @coderabbitai with this change.

**Pull request merged**